### PR TITLE
Trainer only references accelerator

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

There were a few places where the trainer references the plugins directly. Now the trainer only references the accelerator (except for a few edge cases), and the accelerator calls the training type/precision plugins. This is crucial for accelerators that will require custom/non plugin supported behaviour.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/accelerator.py [hunk 1]
```diff
@@ -76,6 +76,26 @@ def setup(self, trainer: "Trainer", model: LightningModule) -> None:
...
+    def pre_dispatch(self) -> None:
+        """Hook to do something before the training/evaluation/prediction starts."""
+        self.training_type_plugin.pre_dispatch()
+        self.precision_plugin.pre_dispatch()
+
+    def post_dispatch(self) -> None:
+        """Hook to do something before the training/evaluation/prediction starts."""
+        self.training_type_plugin.post_dispatch()
+        self.precision_plugin.post_dispatch()
+        self.teardown()
```
Comment (justusschock) [inline pytorch_lightning/accelerators/accelerator.py:97]: Does this have to be part of post-dispatch?

Doesn't the Trainer have a teardown that should/could call this teardown?

**Review by justusschock — APPROVED** [review]
Just one minor question. I like it.

File: pytorch_lightning/accelerators/accelerator.py (hunk 1 above)
Comment (SeanNaren) [reply pytorch_lightning/accelerators/accelerator.py:97]: great point! the trainer should be responsible for calling tear down as its responsible for control over the accelerator, i'll move it

File: pytorch_lightning/accelerators/accelerator.py [hunk 2]
```diff
@@ -397,3 +404,12 @@ def process_dataloader(self, dataloader: Union[Iterable, DataLoader]) -> Union[I
             dataloader: iterable. Ideally of type: :class:`torch.utils.data.DataLoader`
         """
         return self.training_type_plugin.process_dataloader(dataloader)
+
+    @property
+    def results(self) -> Any:
```
Comment (tchaton) [inline pytorch_lightning/accelerators/accelerator.py:408]: Like it !

**Review by tchaton — APPROVED** [review]
LGTM ! Nice cleaning !

File: pytorch_lightning/accelerators/accelerator.py (hunk 1 above)
Comment (justusschock) [reply pytorch_lightning/accelerators/accelerator.py]: @SeanNaren I can also understand having it here, since this is where the accelerators control ends. More a point of discussion I think. 
But to me teardown is more when you leave it finally (we should think about also calling that in the del- method)

File: pytorch_lightning/accelerators/accelerator.py [hunk 3]
```diff
@@ -224,23 +243,6 @@ def validation_step_end(self, output):
         """
         return self.training_type_plugin.validation_step_end(output)
 
-    def predict(self, args):
```
Comment (Borda) [inline pytorch_lightning/accelerators/accelerator.py:227]: is this already in #5951?

File: pytorch_lightning/accelerators/accelerator.py [hunk 4]
```diff
@@ -378,6 +380,10 @@ def on_save(self, checkpoint):
     def barrier(self, name: Optional[str] = None) -> None:
         self.training_type_plugin.barrier(name=name)
 
+    def broadcast(self, obj: object, src: int = 0) -> object:
```
Comment (Borda) [inline pytorch_lightning/accelerators/accelerator.py:383]: what is `src` could be better named or add docsring?

File: pytorch_lightning/trainer/trainer.py [hunk 5]
```diff
@@ -531,26 +531,24 @@ def fit(
 
         self._set_running_stage(None, model)
 
-        return self.training_type_plugin.results or 1
+        return self.accelerator_backend.results or 1
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py:534]: this shall be synced with #6034

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py (hunk 5 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/trainer.py:534]: I will do it  there <3
